packages/core/src/components/Form.js

```diff
@@ -156,7 +156,10 @@ export default class Form extends Component<
       disabled = false
     } = this.props;
     let { fields } = this.state;
-    fields = updateFieldTouchedState(id, true, fields);
+
+    //skiping the field touch update here
+    if(!skipFieldTouched)
```

Comment:
My personal preference here is to always have if blocks within curly brackets. Although this is valid syntax, I think this style is prone to misinterpretation on review. I'd prefer that you add the brackets around the next line

Reply:
Sure I can do that